### ISSUE TITLE
Warning of selflearningcourse is still present without self learning checkbox being checked

### ISSUE THREAD

**Issue by georgmaisser** [Wunderbyte-GmbH/moodle-mod_booking#748]
https://urisepreview.univie.ac.at/mod/booking/editoptions.php?id=1167&optionid=244&returnto=url&returnurl=https%3A%2F%2Furisepreview.univie.ac.at%2Fcourse%2Findex.php%3Fcategoryid%3D6

![Image](https://github.com/user-attachments/assets/789fea9e-17eb-4270-8d37-d318c614a49d)

https://github.com/user-attachments/assets/5673e40b-77cc-4f40-82bf-8e26cc3206b2

**Comment by georgmaisser** [comment]
the reason is that the duration field is not showing up correctly, we need to resave all the config configurations.

**Comment by georgmaisser** [comment]
Fixed with this learning. To explain: new option fields show up from the start (this is intentional, they have to be disabled in the config settings, if not needed). But on "no submit" of a form, the option form configuration is apparently retrieved slightly differently, the new "duration" field was missing. 

The solution was to save all configs with the duration field turned on. (the duration field held a form element which was responsible for hide behaviour of the other fields -> without it, no hide, -> we saw the static warning.
